package/kedro_viz/data_access/repositories/runs.py

```diff
@@ -117,9 +117,16 @@ def create_or_update_user_run_details(
                 user_run_details = UserRunDetailsModel(
                     run_id=run_id, title=title, bookmark=bookmark, notes=notes
                 )
-                session.add(user_run_details)
+                try:
+                    session.add(user_run_details)
+                except Exception as e:
+                    session.rollback()
+                finally:
+                    session.close()
+
```

Comment:
According to my (limited) understanding of sqlalchemy I thought this shouldn't be needed because it's in the `with self._db_session_class.begin()` context manager - why is it needed?